### PR TITLE
Add SROC Supplementary billing acceptance tests

### PR DESCRIPTION
**Pull request by Jozzey**
https://eaflood.atlassian.net/browse/WATER-3980

We need automated acceptance tests to protect the existing functionality from being broken, to avoid increasing the time it takes to sign off a release and to document the behaviours we have discovered and agreed upon.

The following new tests are going to be created in this PR:
- add a new charge version which replaces the previous (no change in billing account or charge reference). This should result in a zero-value bill run
- add a new charge version in the current year which changes the charge reference and adds both adjustments and additional charges

### CODE REVIEW

File: cypress/e2e/internal/billing/supplementary/replace-cv-current-year-no-changes.cy.js [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    cy.get('.govuk-button.govuk-button--start').click()
+
+    //  Assert the user signed in and we're on the search page
+    cy.contains('Search')
+
+    // -------------------------------------------------------------------------
+    cy.log('Replacing initial Charge Version')
+    // There is currently an issue with the initial Charge Version that is created by the service that prevents it from
+    // being replaced with the same billing account, which is essential for this test. As a work around the initial
+    // charge version that was created in the test data is going to be replaced extirely before the initial bill run.
```
Comment (StuAA78) [inline cypress/e2e/internal/billing/supplementary/replace-cv-current-year-no-changes.cy.js]:
```suggestion
    // charge version that was created in the test data is going to be replaced entirely before the initial bill run.
```

File: cypress/e2e/internal/billing/supplementary/replace-cv-current-year-with-changes.cy.js [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    cy.get('.govuk-button.govuk-button--start').click()
+
+    //  Assert the user signed in and we're on the search page
+    cy.contains('Search')
+
+    // -------------------------------------------------------------------------
+    cy.log('Replacing initial Charge Version')
+    // There is currently an issue with the initial Charge Version that is created by the service that prevents it from
+    // being replaced with the same billing account, which is essential for this test. As a work around the initial
+    // charge version that was created in the test data is going to be replaced extirely before the initial bill run.
```
Comment (StuAA78) [inline cypress/e2e/internal/billing/supplementary/replace-cv-current-year-with-changes.cy.js]:
```suggestion
    // charge version that was created in the test data is going to be replaced entirely before the initial bill run.
```

**Review by StuAA78 — CHANGES_REQUESTED** [review]
Small typo

File: cypress/e2e/internal/billing/supplementary/replace-cv-2023-fin-year-no-changes.cy.js [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+      cy.get('div:nth-child(4) > dd').should('contain.text', 'Current')
+    })
+    cy.get(':nth-child(2) > .govuk-grid-column-two-thirds').should('contain.text', '£0.00')
+    cy.get('.govuk-heading-l').should('contain.text', '1 supplementary bill')
+    cy.get('.govuk-table__body > .govuk-table__row > :nth-child(1)').should('contain.text', 'S00000007A')
+    cy.get('.govuk-table__body > .govuk-table__row > :nth-child(2)').should('contain.text', 'Big Farm Co Ltd 02')
+    cy.get('.govuk-table__body > .govuk-table__row > :nth-child(3)').should('contain.text', 'AT/SROC/SUPB/02')
+    cy.currentFinancialYearDate().then((result) => {
+      cy.get('.govuk-table__body > .govuk-table__row > :nth-child(4)').should('contain.text', '2023')
+    })
```
Comment (StuAA78) [inline cypress/e2e/internal/billing/supplementary/replace-cv-2023-fin-year-no-changes.cy.js]: We're not using `result` here so we can get rid of the call to `currentFinancialYearDate()`.
```suggestion
    cy.get('.govuk-table__body > .govuk-table__row > :nth-child(4)').should('contain.text', '2023')
```

**Review by StuAA78 — CHANGES_REQUESTED** [review]
One small change but but all looks good other than that 😁

File: cypress/e2e/internal/billing/supplementary/replace-cv-2023-fin-year-no-changes.cy.js (hunk 3 above)
Comment (Jozzey) [reply cypress/e2e/internal/billing/supplementary/replace-cv-2023-fin-year-no-changes.cy.js]: Well spotted, I'll sort that out

**Review by StuAA78 — APPROVED** [review]

**Pull request merged**